crates/cast/src/script.rs

```diff
@@ -0,0 +1,360 @@
+use cairo_vm::vm::runners::cairo_runner::{ResourceTracker, RunResources};
+use cairo_vm::vm::vm_core::VirtualMachine;
+use camino::Utf8PathBuf;
+use cast::helpers::response_structs::ScriptResponse;
+use cast::helpers::scarb_utils::parse_scarb_metadata;
+use cheatnet::cheatcodes::EnhancedHintError;
+use cheatnet::constants::{build_block_context, build_transaction_context};
+use cheatnet::state::{CheatnetBlockInfo, DictStateReader};
+use clap::command;
+use clap::Args;
+use conversions::StarknetConversions;
+use itertools::chain;
+use num_traits::ToPrimitive;
+use scarb_artifacts::corelib_for_package;
+use starknet::core::types::{BlockId, BlockTag::Pending, FieldElement};
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::JsonRpcClient;
+use tokio::runtime::Runtime;
+
+#[derive(Args)]
+#[command(about = "")]
+pub struct Script {
+    /// Path to the script
+    pub script_path: Utf8PathBuf,
+}
+
+pub struct CairoHintProcessor<'a> {
+    pub blockifier_syscall_handler: SyscallHintProcessor<'a>,
+    pub hints: &'a HashMap<String, Hint>,
+    pub provider: &'a JsonRpcClient<HttpTransport>,
+    pub runtime: Runtime,
+}
+
+// crates/blockifier/src/execution/syscalls/hint_processor.rs:472 (ResourceTracker for SyscallHintProcessor)
+impl ResourceTracker for CairoHintProcessor<'_> {
+    fn consumed(&self) -> bool {
+        self.blockifier_syscall_handler
+            .context
+            .vm_run_resources
+            .consumed()
+    }
+
+    fn consume_step(&mut self) {
+        self.blockifier_syscall_handler
+            .context
+            .vm_run_resources
+            .consume_step();
+    }
+
+    fn get_n_steps(&self) -> Option<usize> {
+        self.blockifier_syscall_handler
+            .context
+            .vm_run_resources
+            .get_n_steps()
+    }
+
+    fn run_resources(&self) -> &RunResources {
+        self.blockifier_syscall_handler
+            .context
+            .vm_run_resources
+            .run_resources()
+    }
+}
+
+impl HintProcessorLogic for CairoHintProcessor<'_> {
+    fn execute_hint(
+        &mut self,
+        vm: &mut VirtualMachine,
+        exec_scopes: &mut ExecutionScopes,
+        hint_data: &Box<dyn Any>,
+        constants: &HashMap<String, Felt252>,
+    ) -> Result<(), HintError> {
+        let maybe_extended_hint = hint_data.downcast_ref::<Hint>();
+        if let Some(Hint::Starknet(StarknetHint::Cheatcode {
+            selector,
+            input_start,
+            input_end,
+            output_start,
+            output_end,
+        })) = maybe_extended_hint
+        {
+            return self.execute_cheatcode_hint(
+                vm,
+                exec_scopes,
+                selector,
+                input_start,
+                input_end,
+                output_start,
+                output_end,
+            );
+        }
+        self.blockifier_syscall_handler
+            .execute_hint(vm, exec_scopes, hint_data, constants)
```

Comment:
I'm not sure about how this should be handled exactly, but I'm not sure if whole dependency on blockifier is needed for just running deployment scripts.

Perhaps https://github.com/starkware-libs/cairo/blob/1c70f6459d1f38018da4397466ba2dccf6d48f81/crates/cairo-lang-runner/src/casm_run/mod.rs#L406 from casm-run would be enough?

Reply:
cc @MaksymilianDemitraszek @Arcticae

Reply:
I guess since deployment scripts are not contracts we can do only this https://github.com/starkware-libs/cairo/blob/1c70f6459d1f38018da4397466ba2dccf6d48f81/crates/cairo-lang-runner/src/casm_run/mod.rs#L417-L419 ?

Reply:
that would be cool, let's try to asses this. If this is doable, and rather quick change, let's do it straight away. If not, let's do this in separate PR. saounds good?

Reply:
It's rather a quick change

Reply:
removed blockifier in https://github.com/foundry-rs/starknet-foundry/pull/1067/commits/9a974fcc0f2ed97df5f152605ac914a5d0a3971d

Reply:
Perfect! Can we clean up the deps in Cargo.toml as well? I believe blockifier is still there

Reply:
good point! 
done